src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax;
+
+import core.basesyntax.dao.ActivityDaoCsvFileImpl;
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.dao.FruitDaoMapImpl;
+import core.basesyntax.dao.ReportDaoCsvFileImpl;
+import core.basesyntax.model.Activity;
+import core.basesyntax.service.ActivityServiceImpl;
+import core.basesyntax.service.ActivityStrategy;
+import core.basesyntax.service.ActivityStrategyImpl;
+import core.basesyntax.service.ReportService;
+import core.basesyntax.service.ReportServiceImpl;
+import core.basesyntax.service.activityhandler.ActivityHandler;
+import core.basesyntax.service.activityhandler.BalanceActivityHandler;
+import core.basesyntax.service.activityhandler.PurchaseActivityHandler;
+import core.basesyntax.service.activityhandler.ReturnActivityHandler;
+import core.basesyntax.service.activityhandler.SupplyActivityHandler;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Main {
+    private static final String INPUT_FILE = "src/main/resources/DailyActivity.csv";
+    private static final String OUTPUT_FILE = "src/main/resources/DailyFruitReport.csv";
+
+    public static void main(String[] args) {
+        FruitDao fruitDao = new FruitDaoMapImpl();
+        Map<Activity.Type, ActivityHandler> operationStrategies = new HashMap<>();
+        operationStrategies.put(Activity.Type.b, new BalanceActivityHandler(fruitDao));
+        operationStrategies.put(Activity.Type.p, new PurchaseActivityHandler(fruitDao));
+        operationStrategies.put(Activity.Type.s, new SupplyActivityHandler(fruitDao));
+        operationStrategies.put(Activity.Type.r, new ReturnActivityHandler(fruitDao));
+
+        ActivityStrategy activityStrategy = new ActivityStrategyImpl(operationStrategies);
+
+        ReportService reportService = new ReportServiceImpl(new ActivityDaoCsvFileImpl(INPUT_FILE),
+                new FruitDaoMapImpl(), new ReportDaoCsvFileImpl(OUTPUT_FILE),
+                new ActivityServiceImpl(activityStrategy));
+        reportService.generateReport();
+    }
```

Comment:
```suggestion
        // call reader and get `List<String>`
        // call parser and get `List<FruitRecord>`
        // call `fruitService.saveData()` - you should pass only `strategy` and `fruitDao` into `FruitServiceImpl` constructor
        // call `fruitService.getFruitReport()` and get String
        // call writer and write the report to the file
```

Reply:
What about the statement that main should only be the entry point to start the program without any logic?

Reply:
currently we will have logic in `main` method
when we will add tests - we will be able to remove code from here